src/nxt_runtime.c

```diff
@@ -797,6 +802,59 @@ nxt_runtime_conf_init(nxt_task_t *task, nxt_runtime_t *rt)
     return NXT_OK;
 }
 
+static nxt_int_t
+nxt_runtime_creds(nxt_task_t *task,  nxt_runtime_t *rt)
+{
+    char          *user, *group_name;
+    struct group  *grp;
+    struct passwd *pwd;
+
+    pwd = getpwuid(geteuid());
+
+    if (pwd == NULL) {
+        user = group_name = NULL;
+    } else {
+        user = pwd->pw_name;
+        grp = getgrgid(pwd->pw_gid);
+
+        if (grp == NULL) {
+            group_name = user;
```

Comment:
This breaks some systems.

For example, if Unit is run by a user with principal gid that is missing in `/etc/group` and there's no group with the same name as user's name. With this change `nxt_user_cred_get()` calls `getgrnam()` and returns `NXT_ERROR` with a log line: `"getgrnam("username") failed, group "username" not found"`. As the result, Unit fails to start.

Previously, it started without a problem, because `nxt_user_cred_get()` doesn't call `getgrnam()` in case when group is empty.

Reply:
@tiago4orion the problem that previously Unit run fine on this system and now it's broken. I propose reverting previous behaviour, where `nxt_user_cred_get()` was called with the group set to NULL.

Reply:
We should handle a case when user's group has no name.

Reply:
Ok, that means we are not going to setgid the processes, ie.: use the same `getegid(2)` for child processes. 

I will leave `group = NULL` in this case and warn the user that the parameter `--group` is being ignored (if unit cannot setgid).

Reply:
@tiago4orion It doesn't mean ignored actually (at least not in all case). Previous behaviour treated special values: `unitd --group ""` or `../configure --group=""` as leaving the group as is. Now it's broken.